serf/snapshot.go

```diff
@@ -320,7 +320,16 @@ func (s *Snapshotter) appendLine(l string) error {
 
 	n, err := s.buffered.WriteString(l)
 	if err != nil {
-		return err
+		wrErr := err
```

Comment:
This feels like it could be pretty aggressive if there are errors for other reasons and lots of writes are being attempted since it'll try to snapshot per line that Serf tries to write. I think we should throttle the recovery attempts to something like 30 seconds. It also might be good to put this logic up up `tryAppend()` since that's kind of focused on handling an append error already, this would be an additional clause after that to check to see if it's time to try to compact and log if that fails.